src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -917,7 +917,9 @@ private static bool HadLambdaConversionError(DiagnosticBag diagnostics, Analyzed
 
             ParameterSymbol parameter = parm == -1 ? null : method.GetParameters()[parm];
             RefKind refArg = arguments.RefKind(arg);
-            RefKind refParm = parameter?.RefKind ?? RefKind.None; // PROTOTYPE: Ref structs?
+            // PROTOTYPE: Resolve the issue of mixed extension overloads (old and new styles) and RefKinds.
+            var isByRef = receiver?.Type?.IsReferenceType == false && System.Linq.Enumerable.All(symbols, m => m.IsInExtensionClass);
```

Comment:
And yeah, perf of linq is definitely not okay here, so I didn't worry about that too much. My diff is just to temporarily fix it, a much bigger change is required to deal with different ref kinds during overload resolution (this is related to my presentation yesterday when I talked about old and new style extension methods having different ref kind receivers)